keps/sig-auth/2579-pod-isolation-policy/README.md

```diff
@@ -0,0 +1,951 @@
+and adding or updating ephemeral containers will require a full policy check.
+
+<<[UNRESOLVED]>>
+
+Once ephemeral containers allow [custom security contexts], it may be desireable to run an ephemeral
+container with higher privileges for debugging purposes. For example, CAP_SYS_PTRACE is forbidden by
+the baseline policy but can be useful in debugging. We could introduce yet-another-mode-label that
+only applies enforcement to ephemeral containers (defaults to the allow policy).
+
+[custom security contexts]: https://github.com/kubernetes/enhancements/tree/master/keps/sig-node/277-ephemeral-containers#configurable-security-policy
+
+<<[/UNRESOLVED]>>
+
+### Pod Security Standards
+
+Policy level definitions are hardcoded and unconfigurable out of the box. However, the [Pod Security
+Standards] leave open ended guidance on a few items, so we must make a static decision on how to
+handle these elements:
+
+**HostPorts** - (baseline) HostPorts will be forbidden. This is a more niche feature, and violates
+the container-host boundary.
+
+**AppArmor** - (baseline) Allow anything except `unconfined`. Custom AppArmor profiles must be
+installed by the cluster admin, and AppArmor fails closed when a profile is not present.
+
+<<[UNRESOLVED]>>
+**SELinux** - (baseline)
+- (Peter Hunt) we could restrict to certain Type values i.e. ban `spc_t` (privileged) or only allow
+  `container_t` (only allowed to touch `container_file_t`, or container contents). That seems
+  consistent with how AppArmor and Seccomp work
+- (Jordan Liggitt) Remember that this admission plugin can't tell the difference between what the
+  user specified and what an earlier mutating admission plugin specified. Disallowing setting
+  selinux labels in the pod spec would also prevent another admission plugin from setting
+  constrained-but-shared-within-the-same-namespace selinux labels on the pod.
+
+<<[/UNRESOLVED]>>
+
+**Non-root Groups** - (restricted) This optional constraint will be omitted from the initial
+implementation.
+
+**Seccomp** - (restricted) Allow anything except `unconfined`. Same reasoning as AppArmor.
+
+### Flexible Extension Support
+
+In order to make it as easy as possible to extend and customize this policy controller, we will
+publish the following tools:
+
+- Library implementation - The admission plugin will be written using reusable library code.
+- Webhook implementation - A standalone webhook implementation will be provided (using the same
+  library code), enabling policy enforcement on older clusters.
+- Thorough testing resources, which can be used to validate 3rd party implementations of the policy
+  spec.
+
+### Test Plan
+
+The admission controller can safely be enabled as a no-op with the default-defaults, i.e. everything
+is privileged. This will let us run the admission controller in our standard E2E test jobs, by
+relabeling specific test namespaces.
+
+**E2E Tests:** The following tests should be added:
```

Comment:
Thinking through conformance, i’m not sure this meets the default bar today (workload enablement) for the mechanism itself (making the labels part of conformance), but the impact of a profile being enforced should definitely be allowed to be tested for conformance (you can remain conformant even if you choose very restrictive workloads).  Organizations that wish to impose additional restrictions on pod creation or to have flexibility on enforcement should be able to disregard this mechanism / substitute their own.  It seems reasonable though that *if* you enable this mechanism, we could test conformance within those definitions, which then benefits both consumers and those who wish to ensure workloads stay consistent.

Reply:
Added an extension section on conformance. Does that address your comments here?

Reply:
Yes.